node/core/runtime-api/src/cache.rs

```diff
@@ -317,7 +317,10 @@ impl RequestResultCache {
 	}
 
 	pub(crate) fn cache_session_info(&mut self, key: SessionIndex, value: Option<SessionInfo>) {
-		self.session_info.insert(key, ResidentSizeOf(value));
+		// only cache Some(SessionInfo)
+		if value.is_some() {
+			self.session_info.insert(key, ResidentSizeOf(value));
```

Comment:
Why do we even cache `Option<SessionInfo>` if we only insert `Some`?

Reply:
Because of the macro it's used in. I've tried to change that, but couldn't figure out how to make it compile w/o changing the macro in 5 minutes, so gave up on that.

Reply:
That should really not be the way to go. If you have problems with this things, you should ask and not release a half backed solution.

Any idea why were we request "invalid session info" aka with a not yet running session index?

Reply:
> If you have problems with this things, you should ask and not release a half backed solution.

I don't view it as a half baked solution. It's possible to refactor the macro code to be more generic, but I'm not sure what would be the benefit of that. It will likely be less readable and this is only "nice to have". Besides that the fix is minimal and easy to backport if needed. 

> Any idea why were we request "invalid session info" aka with a not yet running session index?

Not sure why it happened on Kusama. Note that this was implemented as a response to Rococo issue when we failed to scale decode wrong `SessionInfo` definition, so it returned `None`.